README.md

```diff
@@ -365,4 +365,81 @@ With the v3.1 emended OpenAPI document, `@samchon/openapi` converts it to a migr
 >
 > If directly convert from each version of OpenAPI specification to specific LLM's function calling schema, I have to make much more converters increased by cartesian product. In current models, number of converters would be 12 = 3 x 4.
 >
-> However, if define intermediate schema, number of converters are shrunk to plus operation. In current models, I just need to develop only (7 = 3 + 4) converters, and this is the reason why I've defined intermediate specification. This way is economic.
\ No newline at end of file
+> However, if define intermediate schema, number of converters are shrunk to plus operation. In current models, I just need to develop only (7 = 3 + 4) converters, and this is the reason why I've defined intermediate specification. This way is economic.
+
+
+
+
+## Roadmap
+### Guide Documents
+In here README document, `@wrtnlabs/agent` is introducing its key concepts, principles, and demonstrating some examples. 
+
+However, this contents are not fully enough for new comers of AI Chatbot development. We need much more guide documents and example projects are required for education. We have to guide backend developers to write proper definitions optimized for LLM function calling. We should introduce the best way of multi-agent orchestration implementation.
+
+We'll write such fully detailed guide documents until 2025-03-31, and we will continuously release documents that are in the middle of being completed.
+
+### Playground
+https://nestia.io/chat/playground
+
+I had developed Swagger AI chatbot playground website for a long time ago.
+
+However, the another part obtaining function schemas from TypeScript class type, it is not prepared yet. I'll make the TypeScript class type based playground website by embedding TypeScript compiler (`tsc`).
+
+The new playground website would be published until 2025-03-15.
+
+### WebSocket Module
+```typescript
+import { 
+  IWrtnAgentEvent,
+  IWrtnAgentHeader,
+  IWrtnAgentListener,
+  IWrtnAgentService 
+} from "@wrtnlabs/agent";
+import { Driver, WebSocketConnector } from "tgrid";
+
+const main = async (): Promise<void> => {
+  const connector: WebSocketConnector<
+    IWrtnAgentHeader,
+    IWrtnAgentListener,
+    IWrtnAgentService
+  > = new WebSocketConnector(
+    {
+      locale: "en-US",
+    } satisfies IWrtnAgentHeader,
+    {
+      // EVENT LISTENERS
+      conversate: async (event: IWrtnAgentEvent.IConversate) => {},
+      select: async (event: IWrtnAgentEvent.ISelect) => {},
+      execute: async (event: IWrtnAgentEvent.IExecute) => {},
+      describe: async (event: IWrtnAgentEvent.IDescribe) => {},
+    } satisfies IWrtnAgentService,
+  );
+  await connector.connect("http://localhost:3000/chat");
+
+  // CALL SERVER'S FUNCTION
+  const service: Driver<IWrtnAgentService> = connector.getDriver();
+  await service.conversate("Hello, what you can do?");
+};
+main().catch(console.error);
+```
```

Comment:
In my opinion, this code is very good. However, I think it would be helpful to provide an additional version in a different coding style. One version should be the most minimalistic code possible. Since we are creating open-source software, ease of use is crucial. Just like how Redis allows users to work with it effortlessly using only the get and set methods without deep knowledge of its internal workings, our open-source project should also aim for such simplicity. If the code is too long, it might feel overwhelming for readers.

It would be best to present the code in at least two different styles or focus on making it as simple as possible. Otherwise, we should consider adding more detailed explanations about WebSocketModule to separately describe event management techniques.

Reply:
Have to emend when writing guide documents.